src/test/java/org/commcare/formplayer/tests/CaseClaimNavigationTests.java

```diff
@@ -218,6 +219,40 @@ public void testPostInEntryWithQuery_RelevantFalse() throws Exception {
         verifyNoInteractions(webClientMock);
     }
 
+    @Test
+    public void testNormalClaimWithPostInEntry() throws Exception {
+        ArrayList<String> selections = new ArrayList<>();
+        // It is relevant in the app set up that the selected menu contains a registration form
+        selections.add("3");  // select menu
+        selections.add("1");  // select form
+        selections.add("action 1");  // load case search
+
+        QueryData queryData = new QueryData();
+        queryData.setExecute("case_search.m1_results", true);
+        EntityListResponse entityListResponse;
+        try (MockRequestUtils.VerifiedMock ignore = mockRequest.mockQuery("query_responses/case_claim_response.xml")) {
+            entityListResponse = sessionNavigateWithQuery(selections,
+                    APP_PATH,
+                    queryData,
+                    EntityListResponse.class);
+        }
+        assertEquals(1, entityListResponse.getEntities().length);
+        assertEquals("0156fa3e-093e-4136-b95c-01b13dae66c6",
+                entityListResponse.getEntities()[0].getId());
+
+        selections.add("0156fa3e-093e-4136-b95c-01b13dae66c6");
+        FormEntryResponseBean formEntryResponseBean;
+        try (
+                MockRequestUtils.VerifiedMock ignoredPostMock = mockRequest.mockPost(true, 2);
```

Comment:
why does post happens twice here ?

Reply:
It's because with the configuration
- Menu with case search
- Menu containing both followup and registration form
- sync upon form entry

Then the app work flow would be Select Menu -> select form -> case search query -> select entity -> form entry

Autoadvance session always upon navigate menu. But after case search select entity, `rebuildSessionFromFrame` also runs. So both `autoAdvanceSession` and `rebuildSessionFromFrame` will try to post. Although logic is now added to `doPostAndSync` such that the post does not actually happen twice. But maybe that conditional check should be done by the caller so that `doPostAndSync` does not get called twice?

Reply:
Makes sense, I don't think that will solve the issue since while rebuilding the session we reset the session state which I think will remove the `EXTRA_POST_SUCCESS` as well.  I was thinking that the [relevant condition](https://github.com/dimagi/formplayer/blob/master/src/test/resources/archives/case_claim_post_in_entry/suite.xml#L355) on the post itself should stop the second post from happening again as after the first post, we perform a restore which we are mocking here to include the selected case.

Reply:
By "while rebuilding the session we reset the session state" Do you mean the reset done [here](https://github.com/dimagi/formplayer/blob/0b20bf3c6e87f40ddb331cb8b3ea2f2b9053cc3b/src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java#L92)?

The two calls to `doPostAndSync` happens after that reset so `EXTRA_POST_SUCCESS ` is not removed between the calls. First during the `rebuildSessionFromFrame` and second during `autoAdvanceSession`, I think both are the post related to form entry. This is the logging output
```
   autoadvancing session. screen is: MenuScreen['Case Claim' with choices [Entry with id m3-f0, Entry with id m3- 
    f.1]]
    autoadvancing session. screen is: EntityScreen [id=case_id, selection=null]
    autoadvancing session. screen is: QueryScreen[Case Claim]
    autoadvancing session. screen is: EntityScreen [id=case_id, selection=null]
    autoadvancing session. screen is: org.commcare.formplayer.screens.FormplayerSyncScreen@7a5073f8
    rebuilding session from frame, screen isMenuScreen['Case Claim' with choices [Entry with id m0-f0, Menu with id m1, Menu with id m2, Menu with id m3]]
    rebuilding session from frame, screen isMenuScreen['Case Claim' with choices [Entry with id m3-f0, Entry with id m3-f1]]
    rebuilding session from frame, screen isEntityScreen [id=case_id, selection=null]
    rebuilding session from frame, screen isorg.commcare.formplayer.screens.FormplayerSyncScreen@108eb810
    rebuilding sesssion from frame: screen.getSessionSuccessStatus() is null
    autoadvancing session. screen is: org.commcare.formplayer.screens.FormplayerSyncScreen@108eb810
    autoadvancingsession screen.getSessionSuccessStatus() is true
```

I think the user is being restored since the [mock is called twice](https://github.com/dimagi/formplayer/blob/45e146346aa3060d6a94b9aee2ff6a861ca9a5cb/src/test/java/org/commcare/formplayer/tests/CaseClaimNavigationTests.java#L247)? Also, I think the post we're interested in is the [entry's](https://github.com/dimagi/formplayer/blob/45e146346aa3060d6a94b9aee2ff6a861ca9a5cb/src/test/resources/archives/case_claim_post_in_entry/suite.xml#L345-L347), not the remote request's. If I changed that `relevant` condition to `count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id]) = 0`, then the tests calls `doPostAndSync` only once.

Reply:
> I think the post we're interested in is the [entry's](https://github.com/dimagi/formplayer/blob/45e146346aa3060d6a94b9aee2ff6a861ca9a5cb/src/test/resources/archives/case_claim_post_in_entry/suite.xml#L345-L347), not the remote request's

Ahh yes, I didn't realise we have a post in entry as well. Given that, My sense is the 2 posts that are happening here belongs to 2 different code areas with first one being the remote request one and later being the entry post. I say this because we are adding `action 1` in the selections which points to the remote request with command id `case_search.m1`. This should trigger the first post on case selection and then when the user entes the form we trigger the entry->post as well. This also will explain why you only see 1 post on changing the relevancy condition for entry post. 

Confirming that overall this seems expected behaviour in context of this test.